src/main/java/core/basesyntax/serviceimpl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.serviceimpl;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.service.ReportGenerator;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String REGEX_TO_SPLIT = ",";
+    private static final String CONTENTS = "fruit, quantity";
+
+    @Override
+    public String createReport() {
+        StringBuilder stringBuilder = new StringBuilder(CONTENTS + System.lineSeparator());
+        for (Map.Entry<String, Integer> entry : FruitStorage.fruitRepository.entrySet()) {
+            stringBuilder.append(entry.getKey())
+                    .append(REGEX_TO_SPLIT)
+                    .append(entry.getValue())
+                    .append(System.lineSeparator());
+        }
+        return stringBuilder.toString();
+    }
+}

```

Comment:
It's a good practice to avoid direct interaction with the storage from outside of the db package. Consider creating a service that provides the necessary operations on the FruitStorage and call that service here instead.